docs/source/custom_models.mdx

```diff
@@ -0,0 +1,153 @@
+<!--Copyright 2020 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+-->
+
+# Sharing custom models
+
+The 🤗 Transformers library is designed to be easily extensible. Every model is fully coded in a given subfolder
+of the repository with no abstraction, so you can easily copy a modeling file and tweak it to your needs.
+
+Once you are happy with those tweaks and trained a model you want to share with the community, there are simple steps
+to push on the Model Hub not only the weights of your model, but also the code it relies on, so that anyone in the
+community can use it, even if it's not present in the 🤗 Transformers library.
+
+This also applies to configurations and tokenizers (support for feature extractors and processors is coming soon).
+
+## Sending the code to the Hub
+
+First, make sure your model is fully defined in a `.py` file. It can rely on relative imports to some other files as
+long as all the files are in the same directory (we don't support submodules for this feature yet). For instance,
+let's say you have a `modeling.py` file and a `configuration.py` file in a folder of the current working directory
+named `awesome_model`, and that the modeling file defines an `AwesomeModel`, the configuration file a `AwesomeConfig`.
+
+```
+.
+└── awesome_model
+    ├── __init__.py
+    ├── configuration.py
+    └── modeling.py
+```
+
+The `__init__.py` can be empty, it's just there so that Python detects `awesome_model` can be use as a module.
+Here is an example of what the configuration file could look like:
+
+```py
+from transformers import PretrainedConfig
+
+
+class AwesomeConfig(PretrainedConfig):
+    model_type = "awesome"
+
+    def __init__(self, attribute=1, **kwargs):
+        self.attribute = attribute
+        super().__init__(**kwargs)
+```
+
+and the modeling file could have content like this:
+
+```py
+import torch
+
+from transformers import PreTrainedModel
```

Comment:
does it also work with relative imports? I would imagine lots of people cloning the repo and then just adapting existing models. In this case there would be relative imports here. Could that work?

Reply:
I don't understand what you mean here. Doing a relative import into the Transformers package will fail, as this is not inside the Transformers package anymore. If adapting a model, users will need to replace the imports like
```py
from .file_utils import xxx
```
to
```py
from transformers.file_utils import xxx
```

As seen on the line below, relative imports with other files from that folder works.